CMakeLists.txt

```diff
@@ -68,42 +68,53 @@ if(${BUILD_PYTHON})
     endif()
 endif()
 
-if(MSVC)
-    # TODO: TEST this
-    set(COMPILE_FLAGS  /W4 /WX) 
-else()
```

Comment:
I removed it because I was seeing warnings that were not present when compiling on Linux. These warnings include "possible loss of data during conversion from 'size_t' to 'unsigned int'", "multiple copy constructors specified", "declaration of 'i' hides previous local declaration", "'*/' found outside of comment", etc. 

We may want to look at some of these warnings more closely, but others like "'*/' found outside of comment" are because files like SparseLUFactorization.cpp contain inline "/*  */" style comments. We can add compiler flags to tell Visual Studio to ignore certain kinds of warnings that we don't care about, so it would be possible to add /WX while ignoring unnecessary warnings.

I was thinking of submitting another pull request after this that addresses warnings generated by Visual Studio. Some of the warnings may require code changes that affect building on Linux too, so I didn't want to make this PR too complicated for now and took out the /W4 and /WX flags.